src/wms/src/main/java/org/geoserver/wms/map/GetMapKvpRequestReader.java

```diff
@@ -288,13 +303,19 @@ public GetMapRequest read(Object request, Map kvp, Map rawKvp) throws Exception
         if (layerParam != null) {
             List<String> layerNames = KvpUtils.readFlat(layerParam);
             requestedLayerInfos.addAll(parseLayers(layerNames, remoteOwsUrl, remoteOwsType));
+        } else if (citeCompliant) {
```

Comment:
The SLD extensions to WMS allow a request not to have layers, as long as a full SLD is specified either using &sld or &sld_body. The error must not be thrown in these conditions (which are probably not what INSPIRE had in mind, but nonetheless a OGC specification